src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImportAliasCommand.cs

```diff
@@ -290,103 +378,118 @@ private Collection<AliasInfo> GetAliasesFromFile(bool isLiteralPath)
+            try 
+            {
+                succesfullParse = Enum.TryParse<ScopedItemOptions>(parsedLine[3], out options);
+            } 
+            catch (ArgumentException argException) 
+            {
+                string message = StringUtil.Format(AliasCommandStrings.ImportAliasOptionsError, filePath, lineNumber);
 
-                        FormatException formatException =
-                            new FormatException(message);
+                ErrorRecord errorRecord =
+                    new ErrorRecord(
+                        argException,
+                        "ImportAliasOptionsError",
+                        ErrorCategory.ReadError,
+                        filePath);
 
-                        ErrorRecord errorRecord =
-                            new ErrorRecord(
-                                formatException,
-                                "ImportAliasFileFormatError",
-                                ErrorCategory.ReadError,
-                                filePath);
+                errorRecord.ErrorDetails = new ErrorDetails(message);
+                WriteError(errorRecord);
+            }
 
-                        errorRecord.ErrorDetails = new ErrorDetails(message);
+            return new KeyValuePair<ScopedItemOptions, bool>(options, succesfullParse);
+        }
 
-                        ThrowTerminatingError(errorRecord);
-                    }
+        private AliasInfo ConstructAlias(Collection<string> parsedLine, ScopedItemOptions options) 
+        {
+            AliasInfo newAlias =
+                        new AliasInfo(
+                            parsedLine[0],
+                            parsedLine[1],
+                            Context,
+                            options);
 
-                    ScopedItemOptions options = ScopedItemOptions.None;
+            if (!string.IsNullOrEmpty(parsedLine[2]))
+            {
+                newAlias.Description = parsedLine[2];
+            }
 
-                    try
-                    {
-                        options = (ScopedItemOptions)Enum.Parse(typeof(ScopedItemOptions), values[3], true);
-                    }
-                    catch (ArgumentException argException)
-                    {
-                        string message = StringUtil.Format(AliasCommandStrings.ImportAliasOptionsError, filePath, lineNumber);
+            return newAlias;
+        }
 
-                        ErrorRecord errorRecord =
-                            new ErrorRecord(
-                                argException,
-                                "ImportAliasOptionsError",
-                                ErrorCategory.ReadError,
-                                filePath);
+        private bool IsValidParsedLine(Collection<string> parsedLine, bool optionsParsedSuccesfully, long lineNumber, string filePath) 
+        {
+            if (!optionsParsedSuccesfully)
+            {           
+                return false;
```

Comment:
This place demonstrates that the idea of ​​using `KeyValuePair`  is unjustified.